### PR TITLE
Resolves Open Feedback re: Prohibiting Complex Types

### PR DESCRIPTION
**Pull request by mattmccleary**
## Changes

Resolves open comments from PR (#3858)

1. Clarify the "why" w/o redefining metadata (See [comment](https://github.com/open-telemetry/opentelemetry-specification/pull/3858#discussion_r1508160080))

2. Keep all the new standard attribute content together in one place

This addresses issue (#3925).

### CODE REVIEW

File: specification/common/README.md [hunk 1]
```diff
@@ -98,6 +81,21 @@ reflects that LogRecord attributes are expected to model data produced from
...
+  during several years of OpenTelemetry's development is a useful guardrail for
+  design. In taking additional value types off the table, we narrow the solution
+  space and have more productive design conversations.
+* We proposed extending support for complex value types and received significant
+  pushback. Removing the bounds significantly increases the burden on data
+  consumers. Adding additional simple value types doesn't cause the same level
+  of burden, but these can be encoded using existing primitive types. For
+  example, datetime can be encoded as a string or 64 bit integer.
+* Limiting attribute value types to primitives and arrays of primitives simplifies
+  data consumers' efforts to create search indexes and perform statistical analysis.
```
Comment (jack-berg) [inline specification/common/README.md:97]: 👍 This should have been added before #3858 was merged

File: specification/common/README.md [hunk 2]
```diff
@@ -98,6 +81,21 @@ reflects that LogRecord attributes are expected to model data produced from
 external log APIs, which do not necessarily have the same value type
 restrictions as the standard attribute definition.
 
+**[1]**: NOTE: extending the set of standard attribute value types is a breaking change.
```
[the remark this answers is not in the document]
Comment (mattmccleary) [reply specification/common/README.md:84]: Do we need the footnote? I'd prefer to strike it.
Effectively, the entire "Standard Attributes" section serves as the footnote.

File: specification/common/README.md [hunk 3]
```diff
@@ -98,6 +81,21 @@ reflects that LogRecord attributes are expected to model data produced from
 external log APIs, which do not necessarily have the same value type
 restrictions as the standard attribute definition.
 
+NOTE: extending the set of standard attribute value types is a breaking change.
```
Comment (pellared) [inline specification/common/README.md:84]: nit (this is the most common way of adding a "note" in this repository)

```suggestion
Note: Extending the set of standard attribute value types is a breaking change.
```

**Review by pellared — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

File: specification/common/README.md [hunk 4]
```diff
@@ -98,6 +81,21 @@ reflects that LogRecord attributes are expected to model data produced from
...
+
+* Limiting the types of attribute values to a set which has proved sufficient
+  during several years of OpenTelemetry's development is a useful guardrail for
+  design. In taking additional value types off the table, we narrow the solution
+  space and have more productive design conversations.
+* Upon proposing to extend support for complex value types, we received significant
+  pushback. Limiting attribute value types to primitives and arrays of primitives
+  simplifies data consumers' efforts to create search indexes and perform statistical
+  analysis.
+* To addess concerns over restricting standard attributes to primitive types, it was
```
Comment (pyohannes) [inline specification/common/README.md:95]:
```suggestion
* To address concerns over restricting standard attributes to primitive types, it was
```

**Review by pyohannes — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Comment by carlosalberto** [comment]
Sounds we are good to go. cc @jack-berg

**Review by jack-berg — APPROVED** [review]

**Pull request merged**